domain/relation/service/relation_test.go

```diff
@@ -150,16 +150,18 @@ func (s *relationServiceSuite) TestGetRelationKey(c *gc.C) {
 		ApplicationName: "app-1",
 		Relation: internalcharm.Relation{
 			Name: "fake-endpoint-name-1",
+			Role: internalcharm.RoleProvider,
 		},
 	}
 	endpoint2 := relation.Endpoint{
 		ApplicationName: "app-2",
 		Relation: internalcharm.Relation{
 			Name: "fake-endpoint-name-2",
+			Role: internalcharm.RoleRequirer,
 		},
 	}
 
-	expectedKey := corerelation.Key("app-1:fake-endpoint-name-1 app-2:fake-endpoint-name-2")
+	expectedKey := corerelationtesting.GenNewKey(c, "app-1:fake-endpoint-name-1 app-2:fake-endpoint-name-2")
```

Comment:
todo: This key for test won't `Validate`. The order should be Requirer then Provider.